pyvelox/pyvelox.h

```diff
@@ -105,10 +106,76 @@ inline velox::variant pyToVariant(const py::handle& obj) {
     return pyToVariant<velox::TypeKind::DOUBLE>(obj);
   } else if (py::isinstance<py::str>(obj)) {
     return pyToVariant<velox::TypeKind::VARCHAR>(obj);
+  } else if (py::isinstance<py::list>(obj)) {
+    py::list obj_as_list = py::cast<py::list>(obj);
+    std::vector<velox::variant> result;
+    for (auto& item : obj_as_list) {
+      result.push_back(pyToVariant(item));
+      if (result.front().kind() != result.back().kind()) {
+        throw py::type_error("Array must consist of elements of only one kind");
+      }
+    }
+    return velox::variant::array(std::move(result));
+  } else if (py::isinstance<py::dict>(obj)) {
+    py::dict obj_as_dict = py::cast<py::dict>(obj);
+    std::map<velox::variant, velox::variant> map;
+    for (auto item : obj_as_dict) {
+      velox::variant key = pyToVariant(item.first);
+      velox::variant value = pyToVariant(item.second);
+      map.emplace(std::make_pair(std::move(key), std::move(value)));
+    }
+    return velox::variant::map(std::move(map));
+  } else if (py::isinstance<py::tuple>(obj)) {
+    py::tuple obj_as_tuple = py::cast<py::tuple>(obj);
+    std::vector<velox::variant> elements;
+    elements.reserve(py::len(obj_as_tuple));
+    for (auto item : obj_as_tuple) {
+      elements.emplace_back(pyToVariant(item));
+    }
+    return velox::variant::row(std::move(elements));
   } else {
     throw py::type_error("Invalid type of object");
   }
 }
+template <TypeKind T>
+inline VectorPtr variantToConstantVector(
```

Comment:
You probably need to remerge/rebase this as variantToConstantVector exists in the cpp file : https://github.com/richtia/velox/blob/complex_vectors/pyvelox/pyvelox.cpp#L31

Reply:
ahh...yeah..just removed the one in this PR.